lib/pengine/complex.c

```diff
@@ -248,6 +248,20 @@ get_rsc_attributes(GHashTable *meta_hash, const pcmk_resource_t *rsc,
         get_rsc_attributes(meta_hash, rsc->parent, node, scheduler);
 
     } else {
+        if (pcmk__xe_first_child(scheduler->rsc_defaults,
+                                 PCMK_XE_INSTANCE_ATTRIBUTES, NULL,
+                                 NULL) != NULL) {
+            /* Not possible with schema validation enabled
+             *
+             * @COMPAT Drop support when we can break behavioral
```

Comment:
I don't bother documenting things that only apply with validation disabled. I've never seen it outside of one regression test, and I don't think any users are that foolish.

I'm not sure why we even allow it. Adding custom CIB syntax (without at least adding a custom schema) sounds like a bad idea. :) I suspect it was just convenient in early development. I wonder if we should drop it for 3.0.0. It would remove a lot of problem space for code/schema inconsistencies and simplify some error-checking to assertions.